fsspec/spec.py

```diff
@@ -2040,29 +2096,29 @@ def close(self):
 
         self.closed = True
 
-    def readable(self):
+    def readable(self) -> bool:
         """Whether opened for reading"""
         return self.mode == "rb" and not self.closed
 
-    def seekable(self):
+    def seekable(self) -> bool:
         """Whether is seekable (only in read mode)"""
         return self.readable()
 
-    def writable(self):
+    def writable(self) -> bool:
         """Whether opened for writing"""
         return self.mode in {"wb", "ab"} and not self.closed
 
-    def __del__(self):
+    def __del__(self) -> None:
         if not self.closed:
             self.close()
 
-    def __str__(self):
+    def __str__(self) -> str:
         return f"<File-like object {type(self.fs).__name__}, {self.path}>"
 
     __repr__ = __str__
 
-    def __enter__(self):
+    def __enter__(self) -> "AbstractBufferedFile":
```

Comment:
May want to consider typing_extensions.Self, if it's not already a project dependency, we should add as it's one of the top 10 most popular pypi downloads anyway.